src/Features/Core/Portable/GenerateConstructors/AbstractGenerateConstructorsCodeRefactoringProvider.cs

```diff
@@ -153,26 +155,35 @@ private async Task ComputeRefactoringsAsync(
         Accessibility? desiredAccessibility,
         CancellationToken cancellationToken)
     {
+        // TODO: https://github.com/dotnet/roslyn/issues/5778. Not supported in REPL for now.
+        if (document.Project.IsSubmission)
+            return;
+
         if (document.Project.Solution.WorkspaceKind == WorkspaceKind.MiscellaneousFiles)
-        {
             return;
-        }
 
+        // First, see if we can offer a specific constructor based on what the user has selected, or has their caret on.
         var actions = await GenerateConstructorFromMembersAsync(
             document, textSpan, addNullChecks: false, desiredAccessibility, cancellationToken).ConfigureAwait(false);
-        if (!actions.IsDefault)
+        if (!actions.IsDefaultOrEmpty)
         {
             registerMultipleActions(actions);
         }
-
-        if (actions.IsDefaultOrEmpty && textSpan.IsEmpty)
+        else if (textSpan.IsEmpty)
         {
+            // If that produced nothing, and the user hasn't selected anything explicitly, then also offer the option to
+            // generate a constructor with a dialog.
             var nonSelectionAction = await HandleNonSelectionAsync(document, textSpan, desiredAccessibility, cancellationToken).ConfigureAwait(false);
-            if (nonSelectionAction != null)
-            {
-                registerSingleAction(nonSelectionAction.Value.CodeAction, nonSelectionAction.Value.ApplicableToSpan);
-            }
+            if (nonSelectionAction is var (codeAction, applicableToSpan))
+                registerSingleAction(codeAction, applicableToSpan);
         }
+
+        // Finally, offer to generate default constructors that delegate to the base type if any are missing.
+        var defaultConstructorService = document.GetRequiredLanguageService<IGenerateDefaultConstructorsService>();
+        var defaultConstructorActions = await defaultConstructorService.GenerateDefaultConstructorsAsync(
+            document, textSpan, forRefactoring: true, cancellationToken).ConfigureAwait(false);
```

Comment:
this was all the other refactoring provider did.  just merged it into this shared refactoring provider.